kms/api/Kms.Samples.Tests/KmsFixture.cs

```diff
@@ -0,0 +1,265 @@
+            ParentAsKeyRingName = this.KeyRingName,
+            CryptoKeyId = keyId,
+            CryptoKey = new CryptoKey
+            {
+                Purpose = CryptoKey.Types.CryptoKeyPurpose.AsymmetricSign,
+                VersionTemplate = new CryptoKeyVersionTemplate
+                {
+                    Algorithm = CryptoKeyVersion.Types.CryptoKeyVersionAlgorithm.EcSignP256Sha256,
+                },
+            },
+        };
+        request.CryptoKey.Labels["foo"] = "bar";
+        request.CryptoKey.Labels["zip"] = "zap";
+
+        return client.CreateCryptoKey(request);
+    }
+
+    public CryptoKey CreateAsymmetricSignRsaKey(string keyId)
+    {
+        var client = KeyManagementServiceClient.Create();
+
+        var request = new CreateCryptoKeyRequest
+        {
+            ParentAsKeyRingName = this.KeyRingName,
+            CryptoKeyId = keyId,
+            CryptoKey = new CryptoKey
+            {
+                Purpose = CryptoKey.Types.CryptoKeyPurpose.AsymmetricSign,
+                VersionTemplate = new CryptoKeyVersionTemplate
+                {
+                    Algorithm = CryptoKeyVersion.Types.CryptoKeyVersionAlgorithm.RsaSignPss2048Sha256,
+                },
+            },
+        };
+        request.CryptoKey.Labels["foo"] = "bar";
+        request.CryptoKey.Labels["zip"] = "zap";
+
+        return client.CreateCryptoKey(request);
+    }
+
+    public CryptoKey CreateSymmetricKey(string keyId)
+    {
+        var client = KeyManagementServiceClient.Create();
+
+        var request = new CreateCryptoKeyRequest
+        {
+            ParentAsKeyRingName = this.KeyRingName,
+            CryptoKeyId = keyId,
+            CryptoKey = new CryptoKey
+            {
+                Purpose = CryptoKey.Types.CryptoKeyPurpose.EncryptDecrypt,
+                VersionTemplate = new CryptoKeyVersionTemplate
+                {
+                    Algorithm = CryptoKeyVersion.Types.CryptoKeyVersionAlgorithm.GoogleSymmetricEncryption,
+                },
+            },
+        };
+        request.CryptoKey.Labels["foo"] = "bar";
+        request.CryptoKey.Labels["zip"] = "zap";
+
+        return client.CreateCryptoKey(request);
+    }
+
+    public CryptoKeyVersion CreateKeyVersion(string keyId)
+    {
+        var client = KeyManagementServiceClient.Create();
+
+        var result = client.CreateCryptoKeyVersion(new CreateCryptoKeyVersionRequest
+        {
+            ParentAsCryptoKeyName = new CryptoKeyName(this.ProjectId, this.LocationId, this.KeyRingId, keyId),
+        });
+
+        for (var i = 1; i <= 5; i++)
+        {
+            var version = client.GetCryptoKeyVersion(new GetCryptoKeyVersionRequest
+            {
+                CryptoKeyVersionName = result.CryptoKeyVersionName,
+            });
+
+            if (version.State == CryptoKeyVersion.Types.CryptoKeyVersionState.Enabled)
+            {
+                return version;
+            }
+
+            Thread.Sleep(500 * i);
```

Comment:
This is called relatively infrequently, so I'd prefer to keep it like this for now. If it becomes a performance bottleneck, we can always revisit, but the new KMS tests all run in parallel (because of the splitting into different files) and only take ~10s :smile.